ckan/logic/validators.py

```diff
@@ -902,3 +902,12 @@ def json_object(value):
         raise Invalid(_('Could not parse the value as a valid JSON object'))
 
     return value
+
+
+def extras_valid_json(extras, context):
+    try:
+        for extra, value in iteritems(extras):
+            json.dumps(extra)
```

Comment:
I didn't get it. As i see this construction, for dictionary `{"a": [1,2], "b": {"x": 1}}`, you are converting key to json string without assigning value:
```
extra, value = ("a", [1,2])
json.dumps(extra) == json.dumps("a") == '"a"'

extra, value = ("b", {"x": 1})
json.dumps(extra) == json.dumps("b") == '"b"'
```

 Is `ValueError` ever possible here?

Reply:
ValueError is possible from within ckan or an extension calling the action with something not json compatible, or if you upload a file as an extra field.

Reply:
oh, I see, I should be dumping value not extra, will fix